src/silly_sprites/event_system.cpp

```diff
@@ -0,0 +1,33 @@
+#include "event_system.hpp"
+#include "app_context.hpp"
+
+namespace sly::event {
+    EventHandlerId::EventHandlerId(EventHandlerId&& other) noexcept
+        : m_id{ std::exchange(other.m_id, tl::nullopt) },
+          m_app_context{ std::exchange(other.m_app_context, nullptr) } { }
+
+    EventHandlerId& EventHandlerId::operator=(EventHandlerId&& other) noexcept {;
+        std::swap(m_id, other.m_id);
+        std::swap(m_app_context, other.m_app_context);
+        return *this;
+    }
+
+    EventHandlerId::~EventHandlerId() {
+        if (not m_app_context) {
+            return;
```

Comment:
i dont think we should assert here. it will break if the upcomming memberfunction `.delete()` already defaulted the id.
also i think the event system sould proove that the id has any value. nur the id itself.